RESTserver/src/main/java/com/softserve/if072/restservice/service/HistoryService.java

```diff
@@ -31,35 +35,29 @@
     private String historySuccessfullyOperation;
     @Value("${history.deleteAllSuccessfullyOperation}")
     private String deleteAllSuccessfullyOperation;
+    @Value("${history.containsPages}")
+    private String historyContainsPages;
 
-    public HistoryService(HistoryDAO historyDAO, ForecastService forecastService) {
+    @Autowired
+    public HistoryService(HistoryDAO historyDAO, HistoryRepository historyRepository, ForecastService forecastService) {
         this.historyDAO = historyDAO;
+        this.historyRepository = historyRepository;
         this.forecastService = forecastService;
     }
 
     /**
-     * Make request to a History DAO for retrieving all history records for current user
-     *
-     * @param userID - current user unique identifier
-     * @return list of history records or empty list
-     */
-    public List<History> getByUserId(int userID) {
-        List<History> histories = historyDAO.getByUserId(userID);
-        LOGGER.info(historyContainsRecords, "user", userID, histories.size());
-        return histories;
-    }
-
-    /**
-     * Make request to a History DAO for retrieving all history records by given search fields
+     * Make request to a History repository for retrieving all history records by given search attributes
      *
      * @param userID     - current user unique identifier
      * @param searchData DTO that contains search criterias
-     * @return list of cart records or empty list
+     * @return list of history records or empty list
      */
-    public List<History> getByUserIdAndSearchParams(int userID, HistorySearchDTO searchData) {
-        List<History> histories = historyDAO.searchAllByUserIdAndParams(userID, searchData.getName(),
-                searchData.getDescription(), searchData.getCategoryId(), searchData.getFromDate(), searchData.getToDate());
-        LOGGER.info(historyContainsRecords, "user", userID, histories.size());
+    public Page<History> getByUserIdAndSearchParams(int userID, HistorySearchDTO searchData, Pageable pageable) {
+        Page<History> histories = historyRepository.findByMultipleParams(userID,
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1067)